modules/common/services/audio.nix

```diff
@@ -91,17 +91,70 @@ in
+          name = "initialize-audio-profile";
+          runtimeInputs = [
+            pkgs.pulseaudio
+            pkgs.jq
+          ];
+          text = ''
+            function setProfile() {
+                if [[ $# == 2 ]] && [[ -n "$1" ]] && [[ -n "$2" ]]; then
+                    echo "Setting audio device profile: ($1 - $2)"
+                    pactl set-card-profile "$1" "$2"
+                fi
+            }
+
+            function findAudioProfiles() {
+              local audio_device_name active_profile profile_list
+                if [[ $# == 1 ]] && [[ -n "$1" ]]; then
+                    audio_device_name=$(jq --raw-output --argjson index "$1" '.[$index].name' <<< "$pactl_data")
+                    active_profile=$(jq --raw-output --argjson index "$1" '.[$index].active_profile ' <<< "$pactl_data")
+                    if [[ -n "$audio_device_name" ]]; then
+                        echo "Found the default audio device: $audio_device_name"
+                        profile_list=$(jq --raw-output --argjson index "$1" '.[$index].profiles | keys[]' <<< "$pactl_data")
+                        echo "With profiles: $profile_list"
+                        while IFS= read -r profile; do
+                          setProfile "$audio_device_name" "$profile"
+                        done <<< "$profile_list"
+
+                        echo "Reset the original active profile."
+                        if [[ -n "$active_profile" ]]; then
+                            setProfile "$audio_device_name" "$active_profile"
+                        fi
+                    fi
+                fi
+            }
+
+            pactl_data=$(pactl --format=json list cards)
+            if [[ -z "$pactl_data" ]]; then
+                pulse_address="''${PULSE_SERVER:-localhost}"
+                echo "Error connecting to Pulseaudio service at: \"$pulse_address\""
+                exit 1
+            fi
+
+            default_device_id=$(pactl --format=json info short | jq '.default_sink_name | split(".") | .[1:-1] | join(".")')
+            echo "Default audio device name: $default_device_id";
+
+            audio_device_count=$(jq '. | length' <<< "$pactl_data")
+            echo "Audio device count: $audio_device_count"
+
+            for ((index = 0; index < audio_device_count; index++)); do
+                device_id=$(jq --argjson index $index '.[$index].name | split(".") | .[1:] | join(".")' <<< "$pactl_data")
+                echo "Found audio device with id: $device_id"
+                if [[ "$device_id" == "$default_device_id" ]]; then
+                    findAudioProfiles "$index"
+                fi
+            done
+          '';
+        };
       in
       {
         enable = true;
-        description = "Force selection of Pulseaudio integrated mic profile";
-        path = [ pulse-set-profile ];
+        description = "Ininitialize default audio device profiles";
+        path = [ initialize-audio-profile ];
```

Comment:
Since this is specified, we probably don't need to call the script using the full nix store path in line 166.
It looks like this line should be removed or line 166 changed to `initialize-audio-profile`?

Reply:
Good point, i removed the path arg